### PR TITLE
First PR From Dev to Main

### PR DESCRIPTION
**Pull request by captnjak**

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -1,9 +1,57 @@
...
+const orderProductsRouter = require ('./order_products');
+const productsRouter = require ('./products');
+
+const { getUserById } = require('../db');
 
 apiRouter.get("/", (req, res, next) => {
-  res.send({
-    message: "API is under construction!"
-  });
+    res.status(418);
```
Comment (jdmann) [inline routes/index.js:16]: good status code

File: routes/orders.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    createOrder
+} = require('../db');
+
+// Declarations
+const ordersRouter = express.Router();
+
+
+ordersRouter.get('/', async (req, res, next) => {
+    try {
+        if (req.user) {
```
Comment (jdmann) [inline routes/orders.js:19]: might consider creating a protectedRoute middleware to make this more re-usable

File: routes/products.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+// Declarations
+const productsRouter = express.Router();
+
+productsRouter.get('/', async (req, res, next) => {
+    try {
+        const products = await getAllProducts()
+        if (products) {
+            res.send(products)
+        }
+        next();
```
Comment (jdmann) [inline routes/products.js:22]: else next("error");

Comment (jdmann) [reply routes/products.js:22]: otherwise this may result in a 404, which is not what you want to do, when handling the error

Comment (jdmann) [reply routes/products.js:22]: next is only used for middleware, or handling an error

File: src/components/AllProducts.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	getCurrentUser,
+	storeCurrentUser,
+	getCurrentToken,
+	clearCurrentUser,
+} from '../auth';
+import { Product } from './Product.js';
+
+const AllProducts = ({ products }) => {
+	const [products, setProducts] = useState([]);
+	const handleProducts = async () => {
```
Comment (jdmann) [inline src/components/AllProducts.js:13]: perhaps this should be a useEffect to load the products when the component mounts?

**Review by jdmann — DISMISSED** [review]

**Review by VW71superbeetle — DISMISSED** [review]

**Review by roboMoose — CHANGES_REQUESTED** [review]
Address comments from Jonathan and deploy your Dev branch by Saturday. 
Will do a quick review on Saturday before merging into main

File: routes/users.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+            next({
+                name: 'password-too-short',
+                message: 'Password is too short. 8 or more characters are required. '
+            });
+        } else {
+            const user = await createUser({
+                username,
+                password
+            });
+            console.log("CreateUser Result==>>",user)
```
Comment (roboMoose) [inline routes/users.js:40]: always remove logs!

**Review by roboMoose — APPROVED** [review]
1. make sure codebase clean and tidy.
2. create dummy data and populate db
3. register/login users.

**Pull request merged**